File: poo/java/Professor/ProfessorHorista.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 06/04/14
 *  R.A.: 86492
 */

public class ProfessorHorista extends Professor{

    /* Construtor */
    public ProfessorHorista(String nome, int idade,
                            int matricula, double valor, int horas){
        super(nome, idade, matricula);
        this.valor = valor;
        this.horas = horas;
    }

    /* Getters
    public int getHoras(){
        return this.horas;
    }
    public double getValor(){
        return this.valor;
    }*/

    /* Getter para salario */
    public double getSalario(){
        return this.horas * this.valor;
    }

    /* Setters
    public void setHoras(int horas){
        this.horas = horas;
    }
    public void setValor(double valor){
        this.valor = valor;
    }*/
    
    @Override
    /* Imprime os atributos */
    public void Imprime(){
        super.Imprime();
        System.out.println("\tHoras Trabalhadas:\t" + this.horas +
                           "\n\tValor da Hora:\t" + this.valor +
                           "\n\tSalário:\t" + this.getSalario());
    }

}

File: poo/python/Professor/ProfessorHorista.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 06/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Professor import Professor

class ProfessorHorista(Professor):

    # Construtor #
    def __init__(self, nome, idade, matricula, valor, horas):
        super(ProfessorHorista, self).__init__(nome, idade, matricula)
        self._valor = valor
        self._horas = horas

    # Getter para salario #
    def getSalario(self):
        return self._horas * self._valor

    # Redefinida # 
    # Imprime os atributos #
    def imprime(self):
       super.imprime()
       print('\tHoras Trabalhadas:\t%d' % (self._horas))
       print('\tValor da Hora:\t%.2f' % (self._valor))
       print('\tSalário:\t%.2f' % (self.getSalario()))
